src/pages/components/HistoryList.qml

```diff
@@ -28,6 +28,12 @@ SilicaListView {
         search: view.search
         showDeleteButton: view.showDeleteButton
 
+        function remove() {
+            remorseDelete(function() {
+                view.model.remove(model.index)
```

Comment:
History page's remove button is not using this. Please adjust that as well.

Further, would be better to change history model.remove to be model.url based as that is unique. This way async db && model.index should not be able cause delete for a wrong index. That said, you could store function local variable of the "model.url" in context of the remove function.